crates/rattler_solve/src/libsolv/mod.rs

```diff
@@ -23,22 +24,34 @@ mod wrapper;
 #[derive(Clone)]
 pub struct LibsolvRepoData<'a> {
     /// The actual records after parsing `repodata.json`
-    pub records: &'a [RepoDataRecord],
+    pub records: Vec<&'a RepoDataRecord>,
 
     /// The in-memory .solv file built from the records (if available)
     pub solv_file: Option<&'a LibcByteSlice>,
 }
 
-impl LibsolvRepoData<'_> {
-    /// Constructs a new `LibsolvRepoData` without a corresponding .solv file
-    pub fn from_records(records: &[RepoDataRecord]) -> LibsolvRepoData {
-        LibsolvRepoData {
-            records,
+impl<'a> FromIterator<&'a RepoDataRecord> for LibsolvRepoData<'a> {
+    fn from_iter<T: IntoIterator<Item = &'a RepoDataRecord>>(iter: T) -> Self {
+        Self {
+            records: Vec::from_iter(iter),
             solv_file: None,
         }
     }
 }
 
+impl<'a> LibsolvRepoData<'a> {
+    /// Constructs a new `LibsolvRsRepoData`
+    #[deprecated(since = "0.7.0", note = "use From::from instead")]
```

Comment:
We are at 0.5.0 though (next one will be 0.6, right? 
Happy to deprecate it whenever :) just wanted to note it.

Reply:
Good catch, I mixed it up with pixi.

```suggestion
    #[deprecated(since = "0.6.0", note = "use From::from instead")]
```